starter-core/src/main/java/io/micronaut/starter/feature/graalvm/GraalVMFeatureValidator.java

```diff
@@ -41,8 +41,8 @@ public void validatePostProcessing(Options options, ApplicationType applicationT
                 throw new IllegalArgumentException("GraalVM is not supported in Groovy applications");
             }
 
-            if (options.getJavaVersion().majorVersion() > JdkVersion.JDK_11.majorVersion()) {
-                throw new IllegalArgumentException("GraalVM only supports up to JDK 11");
+            if (options.getJavaVersion().majorVersion() > JdkVersion.JDK_16.majorVersion()) {
+                throw new IllegalArgumentException("GraalVM only supports up to JDK 16");
```

Comment:
We still doesn't support JDK 16 for GraalVM: https://github.com/oracle/graal/issues/3310